xbmc/threads/Condition.h

```diff
@@ -48,8 +69,21 @@ namespace XbmcThreads
       return res == std::cv_status::no_timeout;
     }
 
+    inline void wait(CSingleLock& lock, std::function<bool()> predicate)
+    {
+      cond.wait(lock.get_underlying(), predicate);
+    }
+
     inline void wait(CSingleLock& lock) { wait(lock.get_underlying()); }
 
+    template<typename Rep, typename Period>
```

Comment:
`Rep` stands for Representation?

Reply:
Yep! I tried to use the same nomenclature as the official templates.
```
rep	Rep, an arithmetic type representing the number of ticks
period	Period, a [std::ratio] representing the tick period (i.e. the number of second's fractions per tick)
```
ref: https://en.cppreference.com/w/cpp/chrono/duration